pycbc/distributions/constraints.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2017 Christopher M. Biwer
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+This modules provides classes for evaluating multi-dimensional constraints.
+"""
+
+from pycbc import transforms
+
+class Constraint(object):
+    """ Creates a constraint that evaluates to True if parameters obey
+    the constraint and False if they do not.
+    """
+    name = "custom"
+    required_parameters = []
+    def __init__(self, variable_args, **kwargs):
+
+        # set any given attributes and get transforms from variable_args
+        # to required parameters
+        for kwarg in kwargs.keys():
+            setattr(self, kwarg, kwargs[kwarg])
```

Comment:
I don't like having the func attribute set like this. It took me awhile to figure out how the constraint function below wasn't failing. Just make `func` a required argument. And are any other kwargs needed? If not, just delete this, since it makes it really hard to understand what this class may do.